src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataResourceDeserializerHelpers.cs

```diff
@@ -110,6 +109,44 @@ internal static ODataDeserializerContext GenerateNestedReadContext(ODataNestedRe
             return BuildNestedContextFromCurrentContext(readContext, path);
         }
 
+        // Determines the binding path for an OData Path to a given navigationProperty
+        private static IEdmPathExpression GetBindingPath(Routing.ODataPath path, IEdmNavigationProperty navigationProperty)
```

Comment:
looks like it could be extension method for ODataPath and move it to ODataPathExtensions?